project-base/storefront/components/Blocks/Adverts/Adverts.tsx

```diff
@@ -33,13 +33,17 @@ export const Adverts: FC<AdvertsProps> = ({
     className,
     isSingle,
 }) => {
-    const [{ data: advertsData }] = useAdvertsQuery();
-
-    const filteredAdverts = advertsData?.adverts.filter((advert) => advert.positionName === positionName);
+    const [{ data: advertsData }] = useAdvertsQuery({
+        variables: {
+            categoryUuid: currentCategory?.uuid || null,
+            positionName
+        },
+    });
```

Comment:
Since you have added this, I think that the `shouldBeShown` function and the rest of the display logic can be reduced/removed. WDYT?

Reply:
Good point, removed unnecessary`shouldBeShown` filter and used advert uuid as react key in the `.map()` array rendering.